src/TestExplorer.ts

```diff
@@ -0,0 +1,153 @@
+import * as ko from "knockout";
+import { MessageTypes } from "./Contracts/ExplorerContracts";
+import * as ViewModels from "./Contracts/ViewModels";
+import "../less/hostedexplorer.less";
+import "./Explorer/Menus/NavBar/MeControlComponent.less";
+import { ClientSecretCredential } from "@azure/identity";
+import { CosmosDBManagementClient } from "@azure/arm-cosmosdb";
+import * as msRest from "@azure/ms-rest-js";
+import { DatabaseAccountsGetResponse } from "@azure/arm-cosmosdb/esm/models";
+import { TestExplorerParams } from "./TestExplorerParams";
+
+class CustomSigner implements msRest.ServiceClientCredentials {
+  private token: string;
+  constructor(token: string) {
+    this.token = token;
+  }
+
+  async signRequest(webResource: msRest.WebResourceLike): Promise<msRest.WebResourceLike> {
+    webResource.headers.set("authorization", `bearer ${this.token}`);
+    return webResource;
+  }
+}
+
+class TestExplorer {
+  private notebooksTestRunnerApplicationId: string;
+  private notebooksTestRunnerClientId: string;
+  private notebooksTestRunnerClientSecret: string;
+  private notebooksAccountName: string;
+  private notebooksAccountKey: string;
+  private notebooksAccountSubscriptonId: string;
+  private notebooksAccountResourceGroup: string;
+
+  constructor() {
+    window.onload = () => {
+      this.initTestExplorer();
+    };
+    window.addEventListener("message", this.handleMessage.bind(this), false);
+  }
+
+  private parseUrlParams = (): void => {
+    window.location.search
+      .substr(1)
+      .split("&")
+      .forEach(item => {
+        const tmp = item.split("=");
+        const value = decodeURIComponent(tmp[1]);
+        switch (tmp[0]) {
```

Comment:
If you use the above built in APIs, this should become much simpler. You can call `get` for each key